volmdlr/gmsh_vm.py

```diff
@@ -645,9 +645,15 @@ def check_2d(list_nodes):
         :rtype: bool
         """
 
+        checking = set()
         for node in list_nodes:
-            if node[2] != 0:
-                return False
+            if node[2] == 0:
+                checking.add(True)
+            else:
+                checking.add(False)
+
+        if False in checking:
+            return False
```

Comment:
what it is the difference between the old and new overall functionality of the method?